utilities/src/main/java/io/syndesis/qe/utils/DropBoxUtils.java

```diff
@@ -0,0 +1,92 @@
+package io.syndesis.qe.utils;
+
+import com.dropbox.core.DbxException;
+import com.dropbox.core.DbxRequestConfig;
+import com.dropbox.core.v2.DbxClientV2;
+import com.dropbox.core.v2.files.FileMetadata;
+import com.dropbox.core.v2.files.GetMetadataErrorException;
+import io.syndesis.qe.accounts.Account;
+import io.syndesis.qe.accounts.AccountsDirectory;
+import javassist.tools.rmi.ObjectNotFoundException;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Component;
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Optional;
+
+@Slf4j
+@Component
+public class DropBoxUtils {
+    private final DbxClientV2 client;
+
+    public DropBoxUtils() throws ObjectNotFoundException, DbxException {
+        Optional<Account> optional = AccountsDirectory.getInstance().getAccount("QE Dropbox");
+
+        if (optional.isPresent()) {
+            DbxRequestConfig config = new DbxRequestConfig(optional.get().getProperty("clientIdentifier"));
+            this.client = new DbxClientV2(config, optional.get().getProperty("accessToken"));
+        } else {
+            throw new ObjectNotFoundException("DropBox credentials were not found.");
+        }
+        log.info("DropBox client created, logged as: " + client.users().getCurrentAccount());
+    }
+
+    /**
+     * Check if file with filePath exists - dropbox uses its own file structure
+     *
+     * @param filePath example: filePath = /folder/file.name
+     * @return true if file exists
+     */
+    public boolean checkIfFileExists(String filePath) {
+        boolean found = true;
+        try {
+            client.files().getMetadata(filePath);
+        } catch (GetMetadataErrorException exception) {
+            if (exception.errorValue.isPath() && exception.errorValue.getPathValue().isNotFound()) {
+                found = false;
+            }
+        } catch (DbxException e) {
+            e.printStackTrace();
+            found = false;
+        }
+        return found;
+    }
+
+    /**
+     * Example: deleteFile("/someFolder/file.name")
+     *
+     * @param filePath
+     * @throws DbxException
+     */
+    public void deleteFile(String filePath) throws DbxException {
+        client.files().deleteV2(filePath);
+    }
+
+    /**
+     * Example: uploadFile("/someFolder/file.name", "whatever Text You Want Inside")
+     *
+     * @param filePath
+     * @param text
+     * @throws IOException
+     * @throws DbxException
+     */
+    public void uploadFile(String filePath, String text) throws IOException, DbxException {
+        try {
```

Comment:
I tried it but it says that the code does not support java 1.8

Reply:
Hmm maybe local IDE setting? Because compiler is set for 1.8 https://github.com/syndesisio/syndesis-qe/blob/master/pom.xml#L14-L15

Reply:
I am stupid, I was trying to autoclose temp file instead of streams and ofc it does not support that structure as there is nothing to close...

What about this:
```
        File temp = File.createTempFile(filePath, "");
        try (BufferedWriter bw = new BufferedWriter(new FileWriter(temp))) {
            temp.deleteOnExit();
            bw.write(text);
            bw.flush();
            try (InputStream is = new FileInputStream(temp)) {
                client.files().uploadBuilder(filePath)
                        .uploadAndFinish(is);

                log.debug("Text inside new file: {}", is.toString());
            }
        } catch (IOException ex) {
            log.error("Error with tmp file: " + ex);
        }
```